timemachine/fe/single_topology.py

```diff
@@ -664,9 +664,8 @@ def concatenate(arrays, empty_shape, empty_dtype):
     )
 
     num_atoms = mol_a.GetNumAtoms() + mol_b.GetNumAtoms() - len(core)
-    assert get_num_connected_components(num_atoms, bond_potential.potential.idxs) == 1, (
-        "hybrid molecule has multiple connected components"
-    )
+    if get_num_connected_components(num_atoms, bond_potential.potential.idxs) > 1:
+        warnings.warn("Hybrid molecule has multiple connected components")
```

Comment:
Is there some weaker property than `num_connected_components == 1` we should still assert here?

I don't know what the correct behavior is, but maybe something like "for every `connected_component` in the input topology, assert that `num_connected_components(attach_dumy_atoms(connected_component)) == 1`"

Reply:
I thought about this a bit... and I'm still not sure how to implement something like `attach_dumy_atoms` at the "generic" level if we need to support AHFE. I think having something like this being re-implemented later on specifically for RBFE/RHFE code-paths would make a lot of sense